### PR TITLE
Day2

### PR DESCRIPTION
**Pull request by rogers-razor**

### CODE REVIEW

File: Question2.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	fmt.Scanf("%d %d %d",&rating_module.numberOfStudent,&rating_module.lowestRating,&rating_module.highestRating)
+	var totalRating uint64//atomic counter for accumulating Rating by each student
+	for currentStudent := 1; currentStudent <= rating_module.numberOfStudent; currentStudent++{
+		student.Add(1)
+		go rating_module.takeRating(&student,&totalRating)
+	}
+	student.Wait()
+	fmt.Println("Total Rating after ending process: ",totalRating)
+	rating_module.calculateAverageRating(totalRating)
+	fmt.Println("Total Average Rating after ending process: ",rating_module.calculateAverageRating(totalRating))
```
Comment (Gauravsinghal09) [inline Question2.go:42]: redundant line, line 41 already calculates the average rating

Comment (rogers-razor) [reply Question2.go:42]: Sure!

File: Question2.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	mutex sync.Mutex
+}
+func (rating_module *RatingModule) takeRating(student *sync.WaitGroup,totalRating *uint64){
+	defer student.Done()
+	rating_module.mutex.Lock()
+	rand.Seed(time.Now().UnixNano())
+	currentStudentRating := uint64(rating_module.lowestRating) +
+				uint64(rand.Intn(rating_module.highestRating - rating_module.lowestRating + 1))
+	fmt.Println(currentStudentRating)
+	*totalRating += currentStudentRating
```
Comment (Gauravsinghal09) [inline Question2.go:22]: mutex is required only for this line, would be more efficient

File: Question3.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	for depositId := 1; depositId <= 10; depositId++{
+		waitGroup.Add(1)
+		go accountDetails.deposit(rand.Intn(DepositWithdrawalRange),&waitGroup)
+	}
+	for withdrawalNo := 1; withdrawalNo <= 10; withdrawalNo++{
+		waitGroup.Add(1)
+		go accountDetails.withdraw(withdrawalNo,rand.Intn(DepositWithdrawalRange),&waitGroup,errorChannel)
+	}
+	WorkerChannel := make(chan bool)
+	go errorMessage(errorChannel,WorkerChannel)
```
Comment (Gauravsinghal09) [inline Question3.go:59]: in order to prevent any issues, close the error channel before looping it

File: Question2.go (hunk 2 above)
Comment (rogers-razor) [reply Question2.go:22]: i have used it in my laptop forgot to push will just push it.

File: Question3.go (hunk 3 above)
Comment (rogers-razor) [reply Question3.go:59]: ok will see

File: Question2.go (hunk 2 above)
Comment (rogers-razor) [reply Question2.go:22]: Question 2 with better print statements refer this commit this change is reflected.

File: Question1.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	fmt.Println("{")
+	keys := make([]string, 0, len(frequencyMap))
+	for key := range frequencyMap {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)//sort on the map on the basis of key in frequencyMap
+	for _, key := range keys {
+		fmt.Println(strconv.Quote(key), ":",frequencyMap[key])
+	}
+	fmt.Println("}")
```
Comment (Gauravsinghal09) [inline Question1.go:66]: Try to wrap line 61-65 in a function, as Kunal advised to keep the main function as per the business logic

Comment (rogers-razor) [reply Question1.go:66]: Yeah sure but like the problem mentioned a specific type of format for printing. So I sorted the map on the basis of it's keys and then printed the output.

**Pull request merged**